front/lib/swr/data_source_views.ts

```diff
@@ -211,3 +212,28 @@ export function useDataSourceViewConnectorConfiguration({
     isDocumentError: error,
   };
 }
+
+export function useDataSourceViewTables({
+  dataSourceView,
+  workspaceId,
+}: {
+  dataSourceView: DataSourceViewType | null;
+  workspaceId: string;
+}) {
+  const tablesFetcher: Fetcher<ListTablesResponseBody> = fetcher;
+  const disabled = !dataSourceView;
+
+  const { data, error, mutate } = useSWRWithDefaults(
+    disabled
+      ? null
+      : `/api/w/${workspaceId}/vaults/${dataSourceView.vaultId}/data_source_views/${dataSourceView.sId}/tables`,
+    tablesFetcher
+  );
+
+  return {
+    tables: useMemo(() => (data ? data.tables : []), [data]),
+    isTablesLoading: !error && !data,
```

Comment:
isTablesLoading is going to return `true` when disabled, i suggest `!disabled && !error && !data`